### ISSUE TITLE
SOLVED: Server pull:  Error: Call to a member function json() on array

### ISSUE THREAD

**Issue by skalvaro** [MISP/MISP#9601]
### Actual behavior

When performing a server pull using cake we get an error 'Error: Call to a member function json() on array'.
The job seems to be stuck @ 75% in the jobs list.
When performing a server pull using the gui the result is the same except that the job now returns 'Failed' with the stacktrace available to be viewed.
In both cases the message of the job shows 'Pulling sightings'.

When checking it does seem that pulling the events itself is successful as we can see new events that originated from the pulled server.

### Expected behavior

Server pull executes without errors.

### Steps to reproduce

Update to  v2.4.186 (708d18174af0cf9a38e6358bec10ff24656a86c4) and perform a server pull, either through the gui or via CLI cake Server pull command.

### Version

2.4.186

### Operating System

RedHat

### Operating System version

7.9

### PHP version

7.4

### Relevant log output

```shell
2024-03-01 16:32:39 Error: [JOB ID: 774b86e9-f199-4f56-be52-dea5ac816a22] - failed with error code 1 after 46.612 seconds. STDERR: Error: Call to a member function json() on array
#0 /var/www/MISP/app/Model/Server.php(687): AnalystData->pull(Array, Object(ServerSyncTool))
#1 /var/www/MISP/app/Console/Command/ServerShell.php(149): Server->pull(Array, 'full', Array, '527478', false)
#2 /var/www/MISP/app/Lib/cakephp/lib/Cake/Console/Shell.php(459): ServerShell->pull()
#3 /var/www/MISP/app/Lib/cakephp/lib/Cake/Console/ShellDispatcher.php(222): Shell->runCommand('pull', Array)
#4 /var/www/MISP/app/Lib/cakephp/lib/Cake/Console/ShellDispatcher.php(66): ShellDispatcher->dispatch()
#5 /var/www/MISP/app/Console/cake.php(45): ShellDispatcher::run(Array)
#6 {main}
. STDOUT: .
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by skalvaro** [comment]
@JakubOnderka I can confirm the fix seems to do the trick.
I applied the changes on the two files in the fix on top of  v2.4.186 (708d18174af0cf9a38e6358bec10ff24656a86c4) and the pull job now successfully completes.

**Comment by Kagee** [comment]
2.4.187 is now tagged.

**Comment by skalvaro** [comment]
Solved in v2.4.187
